drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -293,7 +293,7 @@ static int siwx91x_apply_power_save(struct siwx91x_dev *sidev)
 
 	sl_wifi_get_performance_profile(&sl_ps_profile);
 
-	if (sidev->ps_params.enabled == WIFI_PS_DISABLED) {
+	if (sidev->ps_params.enabled == WIFI_PS_DISABLED || sidev->state < WIFI_STATE_ASSOCIATED) {
```

Comment:
`siwx91x_apply_power_save()` is called on join, but is not called on disconnection. Do you know how `HIGH_PERFORMANCE` is applied after a disconnection?